app/models/job_run.rb

```diff
@@ -29,21 +29,14 @@ class JobRun < ApplicationRecord
     end
 
     # signifies the job ran through, but at least one object had an error of some kind (e.g. network blip)
-    event :discovery_report_completed_with_errors do
-      transition running: :discovery_report_complete_with_errors
+    event :completed_with_errors do
```

Comment:
Rather than validate states, make it impossible to end up in a bad state. (Here and below)